src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1051,6 +1060,69 @@ public void flushAndClose() throws IOException {
         }
     }
 
+    protected TranslogRecoveryStats recoverFromTranslog(long translogId, TranslogRecoveryPerformer handler) throws IOException {
```

Comment:
I this should go into the InternalEngine. The shadow engine and a future read only engine don't need this, right?